packages/admin/cms-admin/src/dam/DataGrid/selection/DamMoreActions.tsx

```diff
@@ -114,84 +71,125 @@ export const DamMoreActions = ({ button, transformOrigin, anchorOrigin, folderId
-                        </MenuItem>
-                    </MenuList>
-                </Box>
-            </Menu>
+                {({ handleClose, selectedItemsChip: SelectedItemsChip }) => (
+                    <>
+                        <MoreActionsGroup
+                            groupTitle={<FormattedMessage id="comet.dam.moreActions.overallActions" defaultMessage="Overall actions" />}
+                        >
+                            <MenuItem
+                                disabled={itemsSelected}
+                                onClick={() => {
+                                    folderInputRef.current?.click();
+                                    handleClose();
+                                }}
+                            >
+                                <ListItemIcon>
+                                    <Upload />
+                                </ListItemIcon>
+                                <ListItemText primary={<FormattedMessage id="comet.dam.moreActions.uploadFolder" defaultMessage="Upload folder" />} />
+                            </MenuItem>
+
+                            <MenuItem
+                                disabled={itemsSelected}
+                                onClick={() => {
+                                    editDialogApi.openAddDialog(folderId);
+                                    handleClose();
+                                }}
+                            >
+                                <ListItemIcon>
+                                    <AddFolderIcon />
+                                </ListItemIcon>
+                                <FormattedMessage id="comet.pages.dam.addFolder" defaultMessage="Add Folder" />
+                            </MenuItem>
+                        </MoreActionsGroup>
+
+                        <MoreActionsDivider />
+
+                        <MoreActionsGroup
+                            groupTitle={<FormattedMessage id="comet.dam.moreActions.selectiveActions" defaultMessage="Selective actions" />}
+                        >
+                            {!onlyFoldersSelected && (
+                                <>
+                                    <MenuItem
+                                        onClick={() => {
+                                            handleDownloadClick();
+                                            handleClose();
+                                        }}
+                                    >
+                                        <ListItemIcon>
+                                            <Download />
+                                        </ListItemIcon>
+                                        <ListItemText
+                                            primary={<FormattedMessage id="comet.dam.moreActions.downloadSelected" defaultMessage="Download" />}
+                                        />
+                                        <SelectedItemsChip label={lengthOfSelectedFiles} />
```

Comment:
I'm not sure if I like that this has to be rendered for every item... Did you think about alternatives?

Reply:
We could abstract the `MenuItem` and add a flag and a label prop. But then we have to restrict positioning or add another prop...